clients/common/nm-meta-setting-desc.c

```diff
@@ -3396,7 +3396,6 @@ _set_fcn_ip_config_dns_options (ARGS_SET_FCN)
 
 	nm_assert (!error || !*error);
 
-	nm_setting_ip_config_clear_dns_options (NM_SETTING_IP_CONFIG (setting), TRUE);
```

Comment:
it's correct, but it's rather ugly how this works.

The caller does
```
»···if (modifier != '-') {
»···»···/* Set/add value */
»···»···if (modifier != '+') {
»···»···»···/* We allow the existing property value to be passed as parameter,
»···»···»··· * so make a copy if we are going to free it.
»···»···»··· */
»···»···»···value = value_free = g_strdup (value);
»···»···»···nmc_setting_reset_property (setting, property_name, NULL);
»···»···}
»···»···if (!nmc_setting_set_property (client, setting, property_name, value, &local)) {
```

instead, there should be only one "set_property" function, that gets the modifier. And based on the modifier, the implementation decides whether to append/remove/reset. IOW, set/reset/remove functions should be merged, and act according to an modifier argument.

TODO.